profitbricks-rest/src/test/java/org/apache/jclouds/profitbricks/rest/features/ServerApiLiveTest.java

```diff
@@ -0,0 +1,214 @@
+   public void testListVolumes() {
+      List<Volume> volumes = serverApi().listAttachedVolumes(testServer.dataCenterId(), testServer.id());
+      assertTrue(volumes.isEmpty());
+   }
+   
+   @Test(dependsOnMethods = "testListVolumes")
+   public void testListCdroms() {
+      List<Image> images = serverApi().listAttachedCdroms(testServer.dataCenterId(), testServer.id());
+      assertTrue(images.isEmpty());
+   }
+   
+   @Test(dependsOnMethods = "testListCdroms")
+   public void testAttachCdrom() {
+      attachedCdrom = serverApi().attachCdrom(
+         Server.Request.attachCdromBuilder()
+            .dataCenterId(testServer.dataCenterId())
+            .serverId(testServer.id())
+            .imageId("7cb4b3a3-50c3-11e5-b789-52540066fee9")
+            .build()
+      );
+      assertEquals(attachedCdrom.properties().name(), "ubuntu-14.04.3-server-amd64.iso");
+      assertCdromAvailable(testServer, attachedCdrom.id());
+      
+      List<Image> images = serverApi().listAttachedCdroms(testServer.dataCenterId(), testServer.id());
+      assertEquals(images.size(), 1);
+   }
+   
+   @Test(dependsOnMethods = "testAttachCdrom")
+   public void testRetrieveAttachedCdrom() {
+      Image cdrom = serverApi().getCdrom(testServer.dataCenterId(), testServer.id(), attachedCdrom.id());
+      assertEquals(cdrom.id(), attachedCdrom.id());
+   }
+   
+   @Test(dependsOnMethods = "testRetrieveAttachedCdrom")
+   public void testDetachCdrom() {
+      serverApi().detachCdrom(testServer.dataCenterId(), testServer.id(), attachedCdrom.id());
+      assertCdromRemoved(testServer, attachedCdrom.id());
+   }
+   
+   @Test(dependsOnMethods = "testDetachCdrom")
+   public void testDeleteServer() {
+      serverApi().deleteServer(testServer.dataCenterId(), testServer.id());
+      assertNodeRemoved(ServerRef.builder().dataCenterId(dataCenter.id()).serverId(testServer.id()).build());
+   }
+   
+   private ServerApi serverApi() {
+      return api.serverApi();
+   }
+   
+   private void assertCdromAvailable(Server server, String cdRomId) {
+      assertRandom(new Predicate<String>() {
+         @Override
+         public boolean apply(String args) {
+            String[] params = args.split(",");
+            Image cdrom = serverApi().getCdrom(params[0], params[1], params[2]);
+            
+            if (cdrom == null || cdrom.metadata() == null)
+               return false;
+            
+            return cdrom.metadata().state() == State.AVAILABLE;
+         }
+      }, complexId(server.dataCenterId(), server.id(), cdRomId));
+   }
+
+   private void assertCdromRemoved(Server server, String cdRomId) {
+      assertRandom(new Predicate<String>() {
+         @Override
+         public boolean apply(String args) {
+            String[] params = args.split(",");
+            return serverApi().getCdrom(params[0], params[1], params[2]) == null;
+         }
+      }, complexId(server.dataCenterId(), server.id(), cdRomId));
+   }
+   
+}
```

Comment:
I don't think that we can test volume methods (other than list) in live, because we would need to create or have an existing volume for tests and this is further down the road with the VolumeApi PR.